lms/djangoapps/teams/views.py

```diff
@@ -584,9 +584,23 @@ def delete(self, request, team_id):
         """DELETE /api/team/v0/teams/{team_id}"""
         team = get_object_or_404(CourseTeam, team_id=team_id)
         self.check_object_permissions(request, team)
+        # Note: list() forces the queryset to be evualuated before delete()
+        memberships = list(CourseTeamMembership.get_memberships(team_id=team_id))
```

Comment:
Also, it might be that copying doesn't work, because it's only copying the iterator, not the actual iterator pointers. In that case, `list()` is fine, especially with that comment.